libs/types-common/src/Data/Id.hs

```diff
@@ -47,14 +47,34 @@ data T
 
 data STo
 
+data Mp a
+
+data Op a
+
 type AssetId = Id A
 
 type ConvId = Id C
 
+type MappedConvId = Id (Mp C)
+
+type OpaqueConvId = Id (Op C)
+
 type InvitationId = Id I
 
+-- | A local user ID
 type UserId = Id U
 
+-- | A UUID local to this backend, for which we know a mapping to a
+-- remote qualified user ID exists.
+-- These IDs should never leak to other backends or their clients.
+type MappedUserId = Id (Mp U)
```

Comment:
I'm not sure this type will actually show up in our code, but I'd rather have it here just in case.